xbmc/GUIPassword.cpp

```diff
@@ -106,6 +102,78 @@ bool CGUIPassword::IsItemUnlocked(CFileItem* pItem, const std::string &strType)
         CMediaSourceSettings::GetInstance().Save();
         break;
       }
+      default:
+      {
+        // this should never happen, but if it does, do nothing
+        return false;
+        break;
+      }
+    }
+  }
+  return true;
+}
+
+bool CGUIPassword::IsItemUnlocked(CMediaSource* pItem, const std::string& strType)
+{
+  const std::shared_ptr<CProfileManager> profileManager =
+      CServiceBroker::GetSettingsComponent()->GetProfileManager();
+
+  // \brief Tests if the user is allowed to access the mediasource
+  // \param pItem The mediasource item to access
+  // \param strType The type of share being accessed, e.g. "music", "video", etc. See CSettings::UpdateSources()
+  // \return If access is granted, returns \e true
+  if (profileManager->GetMasterProfile().getLockMode() == LOCK_MODE_EVERYONE)
+    return true;
+
+  while (pItem->m_iHasLock > 1)
+  {
+    std::string strLockCode = pItem->m_strLockCode;
+    std::string strLabel = pItem->strName;
+    int iResult = 0; // init to user succeeded state, doing this to optimize switch statement below
+    char buffer[33]; // holds 32 places plus sign character
+    if (!g_passwordManager.bMasterUser) // Check if we are the MasterUser!
+    {
+      if (0 != CServiceBroker::GetSettingsComponent()->GetSettings()->GetInt(
+                   CSettings::SETTING_MASTERLOCK_MAXRETRIES) &&
+          pItem->m_iBadPwdCount >= CServiceBroker::GetSettingsComponent()->GetSettings()->GetInt(
+                                       CSettings::SETTING_MASTERLOCK_MAXRETRIES))
+      { // user previously exhausted all retries, show access denied error
+        HELPERS::ShowOKDialogText(CVariant{12345}, CVariant{12346});
+        return false;
+      }
+      // show the appropriate lock dialog
+      std::string strHeading = g_localizeStrings.Get(12325);
+      iResult = VerifyPassword(pItem->m_iLockMode, strLockCode, strHeading);
+    }
+    switch (iResult)
+    {
+      case -1:
+      { // user canceled out
+        return false;
+        break;
+      }
+      case 0:
+      {
+        // password entry succeeded
+        pItem->m_iBadPwdCount = 0;
+        pItem->m_iHasLock = 1;
+        g_passwordManager.LockSource(strType, strLabel, false);
+        sprintf(buffer, "%i", pItem->m_iBadPwdCount);
+        CMediaSourceSettings::GetInstance().UpdateSource(strType, strLabel, "badpwdcount", buffer);
+        CMediaSourceSettings::GetInstance().Save();
+        break;
+      }
+      case 1:
+      {
+        // password entry failed
+        if (0 != CServiceBroker::GetSettingsComponent()->GetSettings()->GetInt(
+                     CSettings::SETTING_MASTERLOCK_MAXRETRIES))
+          pItem->m_iBadPwdCount++;
+        sprintf(buffer, "%i", pItem->m_iBadPwdCount);
```

Comment:
This had me wondering about the use the the char[]. Looked into Update source header

https://github.com/xbmc/xbmc/blob/770c691bfeea88d561b39d2e38348ac434db3adc/xbmc/settings/MediaSourceSettings.cpp#L189

Is it maybe worth removing the char[] buffer, using a std::string instead and doing away with the sprintf?

Reply:
i doubt this is jenkins‘ problem. maybe he‘s confused about the hundreds of pushes a made so far...

Reply:
Dont worry about the jenkins failures, we can force a build. Does it work locally for you? If it does, i can kick off a jenkins build

Reply:
yes working locally, please kick off